packages/app-store/routing-forms/lib/findTeamMembersMatchingAttributeLogicOfRoute.ts

```diff
@@ -0,0 +1,480 @@
+    attributesQueryBuilderConfig: attributesQueryBuilderConfig as unknown as Config,
+    isPreview: !!isPreview,
+  });
+
+  if (!logic) {
+    return {
+      teamMembersMatchingAttributeLogic: null,
+      logicBuildingWarnings: null,
+      timeTaken: {
+        ttGetAttributesQueryValue,
+        ttGetAttributesQueryBuilderConfig,
+        ttGetTeamMembersWithAttributeOptionValuePerAttribute,
+      },
+      ...buildTroubleshooterData({
+        type: TroubleshooterCase.NO_LOGIC_FOUND,
+        data: {
+          attributesQueryValue,
+          attributesQueryBuilderConfig,
+          teamMembersWithAttributeOptionValuePerAttribute,
+        },
+      }),
+    };
+  }
+
+  const [
+    { teamMembersMatchingAttributeLogicMap, attributesDataPerUser },
+    ttTeamMembersMatchingAttributeLogic,
+  ] = await aPf(async () =>
+    getLogicResultForAllMembers(
+      {
+        teamMembersWithAttributeOptionValuePerAttribute,
+        attributeJsonLogic: logic,
+        attributesQueryValue,
+      },
+      {
+        concurrency,
+        enableTroubleshooter,
+      }
+    )
+  );
+
+  const teamMembersMatchingAttributeLogic = Array.from(teamMembersMatchingAttributeLogicMap).map((item) => ({
+    userId: item[0],
+    result: item[1],
+  }));
+
+  return {
+    teamMembersMatchingAttributeLogic,
+    logicBuildingWarnings,
+    timeTaken: {
+      ttGetAttributesQueryBuilderConfig,
+      ttGetTeamMembersWithAttributeOptionValuePerAttribute,
+      ttTeamMembersMatchingAttributeLogic,
+      ttGetAttributesQueryValue,
+    },
+    ...buildTroubleshooterData({
+      type: TroubleshooterCase.MATCH_RESULTS_READY,
+      data: {
+        attributesDataPerUser,
+        attributesQueryValue,
+        attributesQueryBuilderConfig,
+        logic,
+        attributesForTeam,
+      },
+    }),
+  };
+
+  function pf<ReturnValue>(fn: () => ReturnValue): [ReturnValue, number | null] {
+    if (!enablePerf) {
+      return [fn(), null];
+    }
+    return perf(fn);
+  }
+  async function aPf<ReturnValue>(fn: () => Promise<ReturnValue>): Promise<[ReturnValue, number | null]> {
+    if (!enablePerf) {
+      return [await fn(), null];
+    }
+    return asyncPerf(fn);
+  }
+}
+
+async function runMainAttributeLogic(data: RunAttributeLogicData, options: RunAttributeLogicOptions) {
+  const { teamMembersMatchingAttributeLogic, ...rest } = await runAttributeLogic(data, options);
+  return {
+    teamMembersMatchingMainAttributeLogic: teamMembersMatchingAttributeLogic,
+    ...rest,
+  };
+}
+
+async function runFallbackAttributeLogic(data: RunAttributeLogicData, options: RunAttributeLogicOptions) {
+  const { teamMembersMatchingAttributeLogic, ...rest } = await runAttributeLogic(data, options);
+  return {
+    teamMembersMatchingFallbackLogic: teamMembersMatchingAttributeLogic,
+    ...rest,
+  };
+}
+
+export async function findTeamMembersMatchingAttributeLogicOfRoute(
+  {
+    form,
+    response,
+    route,
```

Comment:
Now accepts route directly so that in 'Test Preview' we could submit unsaved route as well to be able to test its results.